coderd/database/migrations/000274_check_tags.up.sql

```diff
@@ -0,0 +1,17 @@
+CREATE DOMAIN tagset AS jsonb;
+
+COMMENT ON DOMAIN tagset IS 'A set of tags that match provisioner daemons to provisioner jobs, which can originate from workspaces or templates. tagset is a narrowed type over jsonb. It is expected to be the JSON representation of map[string]string. That is, {"key1": "value1", "key2": "value2"}. We need the narrowed type instead of just using jsonb so that we can give sqlc a type hint, otherwise it defaults to json.RawMessage. json.RawMessage is a suboptimal type to use in the context that we need tagset for.';
```

Comment:
❤️ thanks for adding an explanation.
I'm curious why you needed to add an override for this type instead of doing so on a field-level like with

```yaml
...
          - column: "provisioner_daemons.tags"
            go_type:
              type: "StringMap"
...
```

for example?

Reply:
It's been a few days since I've tested this. Happy to retest. But from memory of when I was generating the code for this, the type inference for the column works just fine.  It just doesn't transitively apply to the parameters passed into the select query.

I'll have a look into this once more just to be able to give you a more concrete answer at least.

Reply:
Absolutely non-blocking; this can be done after merge